### PR TITLE
Neko Project II Kai OPNA emulator

### PR DESCRIPTION
**Pull request by jpcima**
https://github.com/Wohlstand/OPN2BankEditor/issues/46

- emulator + fixes
eliminated warnings, uninitialized variables (too many of them.. worked around by `calloc` hack)
- soft panning
- chip traits support based on chip *family*: OPN2, OPNA
- fixed update of resampler on clock rate changed

### CODE REVIEW

File: src/opnmidi_midiplay.cpp [hunk 1]
```diff
@@ -1211,9 +1212,18 @@ void OPNMIDIplay::noteUpdate(size_t midCh,
...
-#undef BEND_COEFFICIENT
+                double coef;
+                switch (synth.chipFamily()) {
+                    default:
+                    case OPNChip_OPN2:
+                        coef = 321.88557;
+                        break;
+                    case OPNChip_OPNA:
+                        coef = 309.12412;
+                        break;
```
Comment (Wohlstand) [inline src/opnmidi_midiplay.cpp]: Maybe put those coefficients into side of OPN Family template than keep here?

Comment (jpcima) [reply src/opnmidi_midiplay.cpp]: I don't know. I've forgotten how this value is computed, so I scaled it down by a ratio of chip's sampling rates and it worked.
I haven't understood for sure if this constant is more chip-related than it's implementation-related to ADLMIDI, about its value domain for pseudo-frequency values.

Comment (Wohlstand) [reply src/opnmidi_midiplay.cpp]: In a case of here, it's just a constant, so, you can keep it internally, but comment a note about it for just in a case, my mistake I didn't put it into separated place, but just in the code here...

Comment (jpcima) [reply src/opnmidi_midiplay.cpp]: Done. It's the same thing in libADLMIDI and original ADLMIDI, it's a hardcoded value at this place.

**Comment by jpcima** [comment]
Removing some useless crap, macros on attempts to fix Visual Studio...
I don't have this compiler at hand unfortunately..

**Comment by Wohlstand** [comment]
For temporary, I have made a separated bend coefficient for percussions, as because of chip clock difference, percussions are too sensitive and needs a different pitch to sound correctly. Oh, wait! A little idea! I must to add the pitch offset into the bank format!

**Comment by jpcima** [comment]
@Wohlstand it's the sample rate. Why not permit to clock the OPNA at the sample rate desired ? and then to fix the clock at multiple 144 of sample rate.

**Comment by Wohlstand** [comment]
Oh, didn't you made native rate simulation of OPNA?

**Comment by jpcima** [comment]
I does the native and custom rates. I have tried only native, it escaped me the idea of checking this.

**Pull request merged**